relayer/processor/path_end_runtime.go

```diff
@@ -228,108 +299,148 @@ func (pathEnd *pathEndRuntime) shouldSendConnectionMessage(message connectionIBC
 // shouldSendConnectionMessage determines if the channel handshake message should be sent now.
 // It will also determine if the message needs to be given up on entirely and remove retention if so.
 func (pathEnd *pathEndRuntime) shouldSendChannelMessage(message channelIBCMessage, counterparty *pathEndRuntime) bool {
-	msgSendCache, ok := pathEnd.channelSendCache[message.action]
+	action := message.action
+	channelKey := channelInfoChannelKey(message.info).Counterparty()
+	if message.info.Height >= counterparty.latestBlock.Height {
+		pathEnd.log.Debug("Waiting to relay channel message until counterparty height has incremented",
+			zap.Any("channel", channelKey),
+			zap.String("action", action),
+		)
+		return false
+	}
```

Comment:
hi @agouin @mark-rushakoff @jtieri 
Please could you tell me the reason for this check?
Thank you